check_dependency.py

```diff
@@ -0,0 +1,51 @@
+import subprocess
+import sys
+import tkinter as tk
+from tkinter import messagebox
+import webbrowser
+
+def check_java():
+    try:
+        subprocess.run(['java', '-version'], stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+        return True
+    except subprocess.CalledProcessError:
+        return False
+    except FileNotFoundError:
+        return False
+    
+def check_nltk():
+    try:
+        import nltk
+        return True
+    except ImportError:
+        return False
+    
+def prompt_java_installation():
+    root = tk.Tk()
+    root.withdraw()
+    response = messagebox.askyesno("Java Not Found", "Java is not installed. Would you like to install it?")
+    if response:
+        webbrowser.open("https://www.java.com/en/download/")
+    root.destroy()
+
+def prompt_nltk_installation():
+    root = tk.Tk()
+    root.withdraw()
+    response = messagebox.askyesno("NLTK Not Found", "NLTK is not installed. Would you like to install it?")
+    if response:
+        subprocess.check_call([sys.executable, "-m", "pip", "install", "nltk"])
+        root.destroy()
+
+def check_dependency():
+
+    if check_java() == False:
```

Comment:
In check_dependency(), after prompting the user to install Java, I think the script checks for Java again without waiting for installation completion. It would be better to add a more comprehensive way to handle such cases. 
I guess it can be handled by:
- creating loop that can run for a fixed time or until the installation is detected.
Or 
- adding a timeout.

Reply:
How about this instead: install java for them (don't ask them to do the install manually)